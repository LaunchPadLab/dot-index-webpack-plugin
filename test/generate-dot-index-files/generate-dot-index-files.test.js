const path = require('path')
const generateDotIndexFiles = require('../../src/generate-dot-index-files')
const fs = require('fs-extra')

const to = relPath => path.resolve(__dirname, relPath)

test('Requires path argument', () => {
  expect(() => generateDotIndexFiles()).toThrow()
})

test('Generates dot index file correctly', () => {
  generateDotIndexFiles([ to('./test-input') ])
  expect(fs.existsSync(to('./test-input/.index.js'))).toEqual(true)
  const indexFileContent = fs.readFileSync(to('./test-input/.index.js')).toString()
  expect(indexFileContent).toMatchSnapshot()
})

afterAll(() => {
  fs.removeSync(to('./test-input/.index.js'))
})
