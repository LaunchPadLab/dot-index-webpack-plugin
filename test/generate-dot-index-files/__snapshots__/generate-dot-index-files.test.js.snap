// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generates dot index file correctly 1`] = `
"// This index file was auto-generated by dot-index-webpack-plugin.
// To overwrite it, simply add an index.js file of your own.

export { default as MyComponent } from './MyComponent'
export { default as myKebabCasedFunction } from './my-kebab-cased-function'
export { default as myFunction } from './myFunction'
"
`;
